src/main/kotlin/baseball/Application.kt

```diff
@@ -1,5 +1,102 @@
 package baseball
 
+import camp.nextstep.edu.missionutils.Randoms
+import camp.nextstep.edu.missionutils.Console
+
 fun main() {
-    TODO("프로그램 구현")
+    val game = BaseballGame()
+    game.gameStart()
 }
+
+class BaseballGame {
+    private var targetNumbers = mutableListOf<Int>()
+
+    /* 숫자 야구 게임 */
+    fun gameStart() {
+        println("숫자 야구 게임을 시작합니다.")
+        var restartGame = true
+
+        while (restartGame) {
+            targetNumbers.clear()
+            /* 매 게임의 타겟 숫자 3개 선택 */
+            for (i in 0..2)
+                targetNumbers.add(Randoms.pickNumberInRange(1, 9))
+            inputNumbers()
+
+            // 게임 종료
+            restartGame = inputRestart()
+        }
+    }
+
+    /* 게임 도중 3스트라이크가 될 때까지 사용자로부터 3개의 숫자 입력을 받는 함수 */
+    private fun inputNumbers() {
+        var strike = 0
+        var ball = 0
+        
+        while (strike != 3) {
+            print("숫자를 입력해주세요: ")
+            val inputNumbers = Console.readLine()
+
+            /* 예외 처리 */
+            if (inputNumbers.length != 3)
+                throw IllegalArgumentException("길이가 3이 아닌 입력 발생")
+            if (inputNumbers.toIntOrNull() == null)
+                throw IllegalArgumentException("숫자가 아닌 입력 발생")
+            if ('0' in inputNumbers)
+                throw IllegalArgumentException("범위를 벗어난 숫자 입력 발생")
+
+            val result = checkNumbers(inputNumbers)
+            strike = result[0]
+            ball = result[1]
```

Comment:
1줄로 처리 가능하도록 바꿔보시면 좋을 것 같습니다.

```suggestion
val (strike, ball) = checkNumbers(inputNumbers)

```